dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+      for (int i = 0; i < frames.length; i++) {
+        final FrameData frame = frames[i];
+        final String currentCommandType = frame.paths[pathIdx].commands[commandIdx].type;
+        if (commandType != currentCommandType)
+          throw new Exception(
+              'Paths must be built from the same commands in all frames'
+              'command $commandIdx at frame 0 was of type \'$commandType\''
+              'command $commandIdx at frame $i was of type \'$currentCommandType\''
+          );
+        for (int j = 0; j < numPointsInCommand; j++)
+          points[j].add(frame.paths[pathIdx].commands[commandIdx].points[j]);
+      }
+      commands.add(new PathCommandAnimation(commandType, points));
+    }
+
+    final List<double> opacities =
+      frames.map<double>((FrameData d) => d.paths[pathIdx].opacity).toList();
+
+    return new PathAnimation(commands, opacities: opacities);
+  }
+
+  /// List of commands for drawing the path.
+  final List<PathCommandAnimation> commands;
+  /// The path opacity for each animation frame.
+  final List<double> opacities;
+
+  @override
+  String toString() {
+    return 'PathAnimation(commands: $commands, opacities: $opacities)';
+  }
+
+  String toDart() {
+    String result = '${kIndent*2}const _PathFrames(\n';
+    result += '${kIndent*3}opacities: const <double> [\n';
+    for (double opacity in opacities)
+      result +='${kIndent*4}$opacity,\n';
+    result += '${kIndent*3}],\n';
+    result += '${kIndent*3}commands: const <_PathCommand> [\n';
+    for (PathCommandAnimation command in commands)
+      result += command.toDart();
+    result += '${kIndent*3}],\n';
+    result += '${kIndent*2}),\n';
+    return result;
+  }
+}
+
+/// Represents the animation of a single path command.
+class PathCommandAnimation {
+  const PathCommandAnimation(this.type, this.points);
+
+  /// The command type.
+  final String type;
+
+  /// A matrix with the command's points in different frames.
+  ///
+  /// points[i][j] is the i-th point of the command at frame j.
+  final List<List<Point<double>>> points;
+
+  @override
+  String toString() {
+    return 'PathCommandAnimation(type: $type, points: $points)';
+  }
+
+  String toDart() {
+    String dartCommandClass;
+    switch (type) {
+      case 'M':
+        dartCommandClass = '_PathMoveTo';
+        break;
+      case 'C':
+        dartCommandClass = '_PathCubicTo';
+        break;
+      case 'L':
+        dartCommandClass = '_PathLineTo';
+        break;
+      case 'Z':
+        dartCommandClass = '_PathClose';
+        break;
+      default:
+        throw new Exception('unsupported path command: $type');
+    }
+    String result = '${kIndent*4}const $dartCommandClass(\n';
+    for (List<Point<double>> pointFrames in points) {
+      result += '${kIndent*5}const <Offset> [\n';
+      for (Point<double> point in pointFrames)
+        result += '${kIndent*6}const Offset(${point.x}, ${point.y}),\n';
+      result += '${kIndent*5}],\n';
+    }
```

Comment:
bonus points if you can get _PathClose not to have a line break between its `(` and `)` :-)

Reply:
Same as before, I rather keep the code simpler